src/settings.js

```diff
@@ -1453,6 +1453,17 @@ var USE_PTHREADS = 0;
 // [link] - affects generated JS runtime code at link time
 var PTHREAD_POOL_SIZE = '';
 
+// Normally, applications can create new threads even when the pool is empty.
+// When application breaks out to the JS event loop before trying to block on
+// the thread via `pthread_join` or via manual condvars,
+// an extra Worker will be created and the thread callback will be executed.
+// However, breaking out to the event loop requires custom modifications to
+// the code to adapt it to the Web, and not something that works for
+// off-the-shelf apps. Those apps without any modifications are most likely
+// to deadlock. This setting ensures that, instead of a deadlock, they get
+// a runtime error instead that can be at least handled from the JS side.
+var PTHREAD_POOL_SIZE_STRICT = 0;
```

Comment:
I think if we can limit this to requests that come from the main thread we could even make this default to 1?

Reply:
Then developers who know better can set `PTHREAD_POOL_SIZE_STRICT=0` to avoid the warning

Reply:
> I think if we can limit this to requests that come from the main thread we could even make this default to 1?

Not really - as Alon pointed out to me, it's allowed to create pthreads even on the main thread, you just need to return to the event loop before you lock on any shared vars, and apparently that's how some customers use it.

Reply:
What I'm saying is that if we can limit this to only those user who start threads on main (and not from workers or PROZY_TO_PTHREAD users) then maybe we can be OK with asking all the remaining valid users of this feature to opt out of this warning/error (since there may be few enough of them).